### PR TITLE
feat(mimes): Make mimes illiterate

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

I made mimes illiterate .

## Why / Balance

Mimes are supposed to communicate with others through body language and gestures, However due to paper, players have gotten around this limitation by writing what they want to say on paper instead of on chat.

This creates an annoyance for others who are forced to read their writings.
I personally think its quite dumb for a person to choose mime, a character that is not supposed to talk, but talks anyways through paper.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
### Current Version - Shuffles chars
[Screencast from 06-18-2024 06:56:29 PM.webm](https://github.com/space-wizards/space-station-14/assets/103440971/7735160e-114a-491f-b1e6-7cf61221c233)

### First commit - Replaces letters with ?
[Screencast from 06-18-2024 06:30:58 PM.webm](https://github.com/space-wizards/space-station-14/assets/103440971/b4b090dd-8d00-434f-9f7e-d1df3b03b652)

## Breaking changes

**Changelog**

:cl:
- tweak: Mimes are now illiterate. Anything they write will be incomprehensible.

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
Should I make it so they can't read paper either?

**Comment by Brandon-Huu** [comment]
Should I create an  illiterate, give it to mimes, and check for that instead of relying on mimepowers?

**Comment by slarticodefast** [comment]
I like the idea. Monkeys and kobolts should definetely have this without cognizine. Maybe make it so that the letters are scrambled instead of just replacing it with question marks?

**Comment by Simyon264** [comment]
How would mimes fill out paperwork from HoP and such? Just, not?

**Comment by slarticodefast** [comment]
> Should I create an illiterate component, give it to mimes, and check for that instead of relying on mimepowers?

It should be a componenet and trait in my opinion. That way a player can choose if they want it and you can add it to animals.

**Comment by Dutch-VanDerLinde** [comment]
should make them literate whenever they break their vow

**Comment by Brandon-Huu** [comment]
> How would mimes fill out paperwork from HoP and such? Just, not?

HoP would have to follow "the laws" and be more accommodating to those who are not able to communicate in traditional means.

**Comment by Brandon-Huu** [comment]
> should make them literate whenever they break their vow

I can do this.

**Comment by Brandon-Huu** [comment]
> > Should I create an illiterate component, give it to mimes, and check for that instead of relying on mimepowers?
> 
> It should be a component and trait in my opinion. That way a player can choose if they want it and you can add it to animals.

What namespace would the illiterate component go into?

**Comment by beck-thompson** [comment]
Yeah scrambling the letters would probably be better!

**Comment by Brandon-Huu** [comment]
> Yeah scrambling the letters would probably be better!

It scrambles now. 
Added a video to the PR description

**Comment by Dutch-VanDerLinde** [comment]
you also should make this a component instead of hardcoding it to mime powers

File: Content.Server/Paper/PaperSystem.cs [hunk 1]
```diff
@@ -154,6 +155,16 @@ private void OnInputTextMessage(EntityUid uid, PaperComponent paperComp, PaperIn
...
+                if(HasComp<MimePowersComponent>(args.Actor))
+                {
+                  paperComp.Content = new string(
+                      args.Text
+                      .OrderBy(c => Guid.NewGuid())
+                      .ToArray()
+                      );
+                }
+
                 _audio.PlayPvs(paperComp.Sound, uid);
```
Comment (metalgearsloth) [inline Content.Server/Paper/PaperSystem.cs:168]: Tightly coupling this to paper is an awful idea.

Comment (Brandon-Huu) [reply Content.Server/Paper/PaperSystem.cs:168]: Is there a paper event I can use instead?

**Comment by slarticodefast** [comment]
Have you checked what happens with markup text? You might need to sanitize it.
[Here](https://github.com/Moomoobeef/ss14-forms-txt) are some examples you could try.

**Comment by UnicornOnLSD** [comment]
Okay but what if I as a mime enjoy drawing on papers using written description ? that kind of ruins it

I do agree you shouldn't straight up ask things as a mime writing what you want on a paper, but forcing it like that ? Not certain that's the best way

thing I'm talking about:

image

**Comment by luizwritescode** [comment]
Well, as long as mimes can still break their vow using the @laughs command, this is not preventing much. If some mimes don't follow rules why take away the writting priviledge for all mimes? However, I do agree this needs to be a component and a trait for who ever needs it, and built-in for animals without cognize ASAP.

**Comment by UnicornOnLSD** [comment]
> trait for who ever needs it, and built-in for animals without cognize ASAP. Pun Pun shouldn't able to write!

that's an excellent point, it would be great use of that, and make a bit more sense. While pun still could draw ya know, not as well as a mime (or humanoid)

**Comment by UnicornOnLSD** [comment]
Also putting this, trying not to be a bloat for the PR but maybe just add their sensical point. 
image


The trait itself would be great, have use and overall be nice but forced may not be the best bet

**Comment by lizelive** [comment]
part of the problem is that most people have never seen a mime. the biggest problem is 
`job-description-mime = Entertain the crew through non-vocal means, and engage with light rivalry with the clown.`

made a pr #29266 to tweak

**Comment by Brandon-Huu** [comment]
Closed until I finish my PHD. Thesis

**Pull request closed without merging**